src/helper.h

```diff
@@ -317,13 +321,13 @@ class helper_stateful_server : public HelperServerBase
     time_t reservationStart; ///< when the last `reservation` was made
 };
 
-/* helper.c */
-void helperOpenServers(helper * hlp);
-void helperStatefulOpenServers(statefulhelper * hlp);
-void helperSubmit(helper * hlp, const char *buf, HLPCB * callback, void *data);
-void helperStatefulSubmit(statefulhelper * hlp, const char *buf, HLPCB * callback, void *data, uint64_t reservation);
-void helperShutdown(helper * hlp);
-void helperStatefulShutdown(statefulhelper * hlp);
+// TODO: Replace T::Pointers (except those in *Shutdown()?) with T references.
```

Comment:
```suggestion
```

Please remote this TODO:

* helperOpenServers() and helperStatefulOpenServers() share the helper pointer with the started helper processes so we should pass a pointer there, not a reference;
* helperSubmit() and helperStatefulSubmit() have special code for nil helpers; I doubt that code should be there, but refactoring that is outside this PR scope;
* helperShutdown() and helperStatefulShutdown() are similar to C++ delete or clear() -- they should (and do) nothing when there is no helper so that callers do not need to check; their future versions might even set the pointer to nil (like safe_free() does), but that (complex) refactoring is outside this issue scope.